### PR TITLE
sanitize xml from multipart upload in MultiPartTest

### PR DESCRIPTION
**Pull request by lachlan-roberts**
Signed-off-by: Lachlan Roberts <lachlan@webtide.com>

### CODE REVIEW

File: tests/test-webapps/test-servlet-spec/test-spec-webapp/src/main/java/com/acme/test/MultiPartTest.java [hunk 1]
```diff
@@ -67,12 +68,13 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
             for (Part p : parts)
             {
-                out.println("<h3>" + p.getName() + "</h3>");
+                out.println("<h3>" + StringUtil.sanitizeXmlString(p.getName()) + "</h3>");
                 out.println("<b>Size:</b>&nbsp;" + p.getSize());
                 if (p.getContentType() == null || p.getContentType().startsWith("text/plain"))
                 {
                     out.println("<p>");
-                    IO.copy(p.getInputStream(), out);
+                    String content = IO.toString(p.getInputStream());
```
Comment (lachlan-roberts) [inline tests/test-webapps/test-servlet-spec/test-spec-webapp/src/main/java/com/acme/test/MultiPartTest.java]: could this be a problem with a large content size

Comment (gregw) [reply tests/test-webapps/test-servlet-spec/test-spec-webapp/src/main/java/com/acme/test/MultiPartTest.java]: Yes - we should probably limit it echoing the first 32KB (ie size of response buffer).

File: tests/test-webapps/test-servlet-spec/test-spec-webapp/src/main/java/com/acme/test/MultiPartTest.java [hunk 2]
```diff
@@ -67,12 +68,13 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
             out.println("<b>Parts:</b>&nbsp;" + parts.size());
             for (Part p : parts)
             {
-                out.println("<h3>" + p.getName() + "</h3>");
+                out.println("<h3>" + StringUtil.sanitizeXmlString(p.getName()) + "</h3>");
```
Comment (gregw) [inline tests/test-webapps/test-servlet-spec/test-spec-webapp/src/main/java/com/acme/test/MultiPartTest.java]: +1

File: tests/test-webapps/test-servlet-spec/test-spec-webapp/pom.xml [hunk 3]
```diff
@@ -225,5 +225,10 @@
       <!-- deliberately old version to test classloading -->
       <version>9.3.0.RC0</version>
     </dependency>
+    <dependency>
+      <groupId>org.eclipse.jetty</groupId>
+      <artifactId>jetty-util</artifactId>
+      <version>${project.version}</version>
```
Comment (gregw) [inline tests/test-webapps/test-servlet-spec/test-spec-webapp/pom.xml]: Note the dependency above about deliberately using an old version of jetty-util!

Comment (gregw) [reply tests/test-webapps/test-servlet-spec/test-spec-webapp/pom.xml]: Hmmm if sanitize in not in the old version you may have to inline it.

Comment (joakime) [reply tests/test-webapps/test-servlet-spec/test-spec-webapp/pom.xml]: Perhaps we need a standalone war / webapp that specifically tests old versions of jetty-util
one that isn't part of demo-base.
one that is only used during integration testing in the build.

**Pull request merged**